common/src/main/java/io/pravega/common/util/CompositeByteArraySegment.java

```diff
@@ -183,6 +185,31 @@ public CompositeArrayView slice(int offset, int length) {
         assert length == 0 : "Collection finished but " + length + " bytes remaining";
     }
 
+    @Override
+    public Iterator<ByteBuffer> iterateBuffers() {
+        if (this.length == 0) {
+            return Collections.emptyIterator();
+        }
```

Comment:
Does is make sense to have a simpler handling in case of wrapping only 2 buffers?
Probably we can save a few cpu cycles and memory allocations, but I am not sure it is worth

Reply:
I tried to optimize both this and `CompositeBufferView` based on when they have only one (or two) buffers. The difference I saw was measured in fractions of nanoseconds so not worth it in my POV.

FYI, `CompositeByteArraySegment` is only used in one place (currently). When we generate writes to BookKeeper, we need to write at most 1MB, but we do not know how much in advance to allocate. To give the impression of an array, this class creates a "sparse array" made of smaller arrays which are only allocated when first written to. 

Most of the time this will have just 1 element (up to 128KB).